x-pack/plugins/monitoring/public/application/setup_mode/setup_mode.tsx

```diff
@@ -0,0 +1,200 @@
+      body: JSON.stringify({
+        ccs,
+      }),
+    });
+    return response;
+  } catch (err) {
+    // TODO: handle errors
+    throw new Error(err);
+  }
+};
+
+const notifySetupModeDataChange = () => setupModeState.callback && setupModeState.callback();
+
+export const updateSetupModeData = async (uuid?: string, fetchWithoutClusterUuid = false) => {
+  const data = await fetchCollectionData(uuid, fetchWithoutClusterUuid);
+  setupModeState.data = data;
+  const hasPermissions = get(data, '_meta.hasPermissions', false);
+  if (!hasPermissions) {
+    let text: string = '';
+    if (!hasPermissions) {
+      text = i18n.translate('xpack.monitoring.setupMode.notAvailablePermissions', {
+        defaultMessage: 'You do not have the necessary permissions to do this.',
+      });
+    }
+
+    Legacy.shims.toastNotifications.addDanger({
+      title: i18n.translate('xpack.monitoring.setupMode.notAvailableTitle', {
+        defaultMessage: 'Setup mode is not available',
+      }),
+      text,
+    });
+    return toggleSetupMode(false);
+  }
+  notifySetupModeDataChange();
+
+  const clusterUuid = globalState.cluster_uuid;
+  if (!clusterUuid) {
+    const liveClusterUuid: string = get(data, '_meta.liveClusterUuid');
+    const migratedEsNodes = Object.values(get(data, 'elasticsearch.byUuid', {})).filter(
+      (node: any) => node.isPartiallyMigrated || node.isFullyMigrated
+    );
+    if (liveClusterUuid && migratedEsNodes.length > 0) {
+      setNewlyDiscoveredClusterUuid(liveClusterUuid);
+    }
+  }
+};
+
+export const hideBottomBar = () => {
+  setupModeState.hideBottomBar = true;
+  notifySetupModeDataChange();
+};
+export const showBottomBar = () => {
+  setupModeState.hideBottomBar = false;
+  notifySetupModeDataChange();
+};
+
+export const disableElasticsearchInternalCollection = async () => {
+  const clusterUuid = globalState.cluster_uuid;
+  const url = `../api/monitoring/v1/setup/collection/${clusterUuid}/disable_internal_collection`;
+  try {
+    const response = await httpService.post(url);
+    return response;
+  } catch (err) {
+    // TODO: handle errors
+    throw new Error(err);
+  }
+};
+
+export const toggleSetupMode = (inSetupMode: boolean) => {
+  setupModeState.enabled = inSetupMode;
+  globalState.inSetupMode = inSetupMode;
+  globalState.save?.();
+  setSetupModeMenuItem();
+  notifySetupModeDataChange();
+
+  if (inSetupMode) {
+    // Intentionally do not await this so we don't block UI operations
+    updateSetupModeData();
+  }
+};
+
+export const setSetupModeMenuItem = () => {
+  if (isOnPage('no-data')) {
+    return;
+  }
+
+  const enabled = !globalState.inSetupMode;
+  const I18nContext = Legacy.shims.I18nContext;
+
+  render(
+    <KibanaContextProvider services={Legacy.shims.kibanaServices}>
+      <I18nContext>
+        <SetupModeEnterButton enabled={enabled} toggleSetupMode={toggleSetupMode} />
+      </I18nContext>
+    </KibanaContextProvider>,
+    document.getElementById('setupModeNav')
```

Comment:
Why do we renter into the div here rather than put a conditional component into the PageTemplate?

Also I noticed we now have two "setup" things, so guessing we should maybe move the `setupModeNav` into `MonitoringToolbar`?

![Screen Shot 2021-09-03 at 13 38 23](https://user-images.githubusercontent.com/690/131951286-d3890e5b-b20a-49ac-9a1c-d533f6a0308f.png)

Reply:
This is how it was done before and I don't know exactly why they did it that way.

When I added it, it just worked so I thought that changing it will take more time and it could introduce some bugs. I also think that everything related to setup mode needs to be refactored, both from a UX perspective (I'm not sure how intuitive it is for users...) and technically so we could improve everything together.